src/components/v2/V2Project/LaunchProjectPayerModal.tsx

```diff
@@ -0,0 +1,150 @@
+ */
+const getProjectPayerAddressFromReceipt = (
+  txReceipt: TransactionReceipt,
+): string => {
+  const newProjectPayerAddress = txReceipt?.logs[DEPLOY_EVENT_IDX]?.address
+  return newProjectPayerAddress
+}
+export default function LaunchProjectPayerModal({
+  visible,
+  onClose,
+  useDeployProjectPayerTx,
+  onConfirmed,
+}: {
+  visible: boolean
+  onClose: VoidFunction
+  useDeployProjectPayerTx: () => TransactorInstance<{}> | undefined
+  onConfirmed?: VoidFunction
+}) {
+  const [loadingProjectPayer, setLoadingProjectPayer] = useState<boolean>()
+  const [transactionPending, setTransactionPending] = useState<boolean>()
+  const [projectPayerAddress, setProjectPayerAddress] = useState<string>()
+
+  const [confirmedModalVisible, setConfirmedModalVisible] = useState<boolean>()
+  // TODO: load project payer and show different thing in this section if the project already has one
+  // (Issue: #897)
+
+  const deployProjectPayerTx = useDeployProjectPayerTx()
+
+  function deployProjectPayer() {
+    if (!deployProjectPayerTx) return
+
+    setLoadingProjectPayer(true)
+
+    deployProjectPayerTx(
+      {
+        args: [],
+      },
+      {
+        onDone() {
+          setTransactionPending(true)
+        },
+        async onConfirmed(result) {
+          const txHash = result?.transaction?.hash
+          if (!txHash) {
+            return
+          }
+
+          const txReceipt = await readProvider.getTransactionReceipt(txHash)
+          const newProjectPayerAddress =
+            getProjectPayerAddressFromReceipt(txReceipt)
+          if (newProjectPayerAddress === undefined) {
+            return
+          }
+          if (onConfirmed) onConfirmed()
+          onClose()
+          setProjectPayerAddress(newProjectPayerAddress)
+          setLoadingProjectPayer(false)
+          setTransactionPending(false)
+          setConfirmedModalVisible(true)
+        },
+      },
+    )
+  }
+
+  return (
+    <>
+      <TransactionModal
+        visible={visible}
+        title={t`Create payable address`}
+        okText={t`Deploy project payer contract`}
+        onOk={deployProjectPayer}
+        onCancel={() => onClose()}
+        confirmLoading={loadingProjectPayer}
+        transactionPending={transactionPending}
+      >
+        <p>
+          <Trans>
+            Create an ETH address that people can use to pay this project rather
+            than paying through the juicebox.money interface.
+          </Trans>
+        </p>
+        <p>
+          <Trans>
+            Tokens minted from payments to this address will belong to the
+            payer. However, if someone pays the project from a non-custodial
+            entity like the Coinbase app,{' '}
+            <strong>
+              tokens can't be issued to their personal wallets and will be lost
+            </strong>
+            .
+          </Trans>
+        </p>
+        <p>
+          <Trans>
+            This deploys a contract to the blockchain, and will incur
```

Comment:
nit: `This deploys a contract to the blockchain and will incur significant gas fees.` drop the `,`

Reply:
100%